src/qz/ui/IconCache.java

```diff
@@ -231,15 +236,73 @@ public void setBgColor(Icon i, Color bgColor) {
     }
 
     /**
-     * Inverts the colors for a particular icon
-     * For integration with dark or light desktop themes
-     *
-     * @param i the IconCache.Icon
+     * Replaces the cached tray icons with corrected versions if necessary
+     * e.g.
+     *  - Ubuntu transparency
+     *  - macOS masked icons
+     *  - macOS 10.14+ dark mode support
      */
-    public void invertColors(Icon i) {
+    public void fixTrayIcons(boolean darkMode) {
+        // Fix the tray icon to look proper on Ubuntu
+        if (SystemUtilities.isUbuntu()) {
+            UbuntuUtilities.fixTrayIcons(this);
+        }
+
+        // Handle mask-style tray icons
+        if (SystemUtilities.prefersMaskTrayIcon()) {
+            // Clone the mask icon
+            for (String id : Icon.MASK_ICON.getIds()) {
+                BufferedImage src = images.get(id);
+                BufferedImage dest = clone(src);
+                if (darkMode) {
+                    dest = ColorUtilities.invert(dest);
+                }
+                images.put(id.replaceAll("mask", "default"), dest);
+                imageIcons.put(id.replaceAll("mask", "default"), new ImageIcon(dest));
+            }
+        }
+
+        // Handle undocumented macOS tray icon padding
+        for(IconCache.Icon i : IconCache.getTypes()) {
+            // See also JXTrayIcon.getSize()
+            if (i.isTrayIcon() && SystemUtilities.isMac()) {
+                // Prevent padding from happening twice on WARNING_ICON, DANGER_ICON
+                if (!i.padded || i == Icon.DEFAULT_ICON) {
+                    padIcon(i, 25);
+                }
+            }
+        }
+    }
+
+    public static BufferedImage clone(BufferedImage src) {
+        Image tmp = src.getScaledInstance(src.getWidth(), src.getHeight(), src.getType());
+        BufferedImage dest = new BufferedImage(tmp.getWidth(null), tmp.getHeight(null), BufferedImage.TYPE_INT_ARGB);
+        Graphics g = dest.createGraphics();
+        g.drawImage(tmp, 0, 0, null);
+        g.dispose();
+        return dest;
+    }
+
+    public void padIcon(Icon i, int percent) {
         for (String id : i.getIds()) {
-            images.put(id, ColorUtilities.invert(images.get(id)));
-            imageIcons.put(id, new ImageIcon(images.get(id)));
+            // Calculate padding percentage
+            int w = images.get(id).getWidth();
+            int wPad = (int)((percent/100.0) * w);
+            w += wPad;
```

Comment:
Another clarity item, instead of adding to `w` beforehand like this, can we use `w + wPad` in the buffered image arguments? (and for `h` too)

Reply:
Done via bd4937e.